router/src/infer.rs

```diff
@@ -88,6 +90,10 @@ impl Infer {
             .chat_template
             .map(|t| ChatTemplate::new(t, tokenizer_config.bos_token, tokenizer_config.eos_token));
 
+        let completion_template = tokenizer_config
+            .completion_template
```

Comment:
There is if we want the API to support Fill in the Middle: https://github.com/huggingface/text-generation-inference/pull/1527#issuecomment-1930379359

Reply:
Except nothing in HF ecosystem supports this. Therefore it doesn't exist.
And if we added such a thing we'd commit to it, which is not the appropriate place or project for it (we usually start to implement things like this in `transformers` first, since it will have to check for much wider usability).

Reply:
The primary use case across HF would probably be this API. 

Another potential larger use case is I can see how there also could be a utility method which enables "easy" token encoding that would enable developers to send in a prefix & suffix and get back a list of tokens which has been properly created with FIM tokens (as a bonus it could also add the BOS and EOS tokens as well). 

Alternatively, if we want to support FIM models with this API (we sorta do if we want it to be most useful), we could *only* store the per model chat template in this project, not in `tokenizer_config.json` and worry about moving it out of the project later. So this would mean having a configuration file in this project which has a mapping from model -> chat template.

Reply:
Understood! It does mean this API will be less useful for FIM models which means it will not be a drop-in replacement for some (possibly many?) tools built on the legacy OpenAI completions API. This is cumbersome when you don't control the source of those tools but desire to move off of OpenAI to your own hosting.

IMO this defeats most of the value prop for the API -- but I don't have insight into all the internal reasoning.

Reply:
Understood.